app/controllers/application_controller.rb

```diff
@@ -6,6 +6,15 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   after_action :allow_iframe!
 
+  rescue_from CanCan::AccessDenied do |_exception|
+    if signed_in?
+      redirect_to permission_denied_path
```

Comment:
I think this will blow up as there is no route for permission_denied_path, that is something that is provided by devise which isn't being used here